optimum/intel/ipex/inference.py

```diff
@@ -20,7 +26,25 @@
     import intel_extension_for_pytorch as ipex
 
 
-class _ModelFallbackWrapper:
+def prepare_jit_inputs(model):
+    generation_config = GenerationConfig(config=model.config, task="text-generation")
+    input_config = {
+        "input_ids": {"batch_size": 1, "sequence_length": 1},
+        "past_key_values": {"batch_size": 1, "sequence_length": 1},
+        "attention_mask": {"batch_size": 1, "sequence_length": 2},
+    }
+
+    jit_inputs = generation_config.generate_dummy_inputs(framework="pt", **input_config)
+    jit_inputs["past_key_values"] = tuple(jit_inputs["past_key_values"])

```

Comment:
In my previous comment I was suggesting to use directly the onnx config allowing to have the inputs directly generated for a lot of architectures, was this suggestion preferred in order to  not have to add a dependency to onnx ? (which is not necessary in this case I agree but this would makes things much easier IMO)

```python
from optimum.exporters.tasks import TasksManager

onnx_config_class = TasksManager.get_exporter_config_constructor(
    exporter="onnx",
    model=model,
    task=task,
)
dummy_inputs = onnx_config_class.generate_dummy_inputs(framework="pt")
```

with task being the task being defined in https://huggingface.co/docs/optimum/main/en/exporters/task_manager#pytorch which can be inferred such as :
```python
task = TasksManager.infer_task_from_model(model_id)
```

Reply:
The onnx config class may not meet my demand since I need ('input_ids', 'past_key_values', 'attention_masks') in the inputs. BTW, I running the previous code and got the error: `AttributeError: 'functools.partial' object has no attribute 'generate_dummy_inputs'`

Reply:
@echarlaix Hi, Ella. I have tried `OnnxConfig` but it doesn't work. When I run the previous codes, I got the error: `AttributeError: 'functools.partial' object has no attribute 'generate_dummy_inputs'`. When I run `OnnxConfig(model.config, task)`, I got this error: `TypeError: Can't instantiate abstract class OnnxConfig with abstract method inputs`.

Reply:
Apologies @jiqing-feng, you also need to create an instance of OnnxConfig, here is the updated code snippet :

```python
from optimum.exporters.tasks import TasksManager

onnx_config_class = TasksManager.get_exporter_config_constructor(
    exporter="onnx",
    model=model,
    task=task,
)
onnx_config = onnx_config_class(model.config)
dummy_inputs = onnx_config.generate_dummy_inputs(framework="pt")
```
Note : if `task="causal-lm"`, the `past_key_values` will not be generated. For that, you need to have  `task="causal-lm-with-past"` instead.

Also, you can also get the task from pipelines using the `task` attribute, when using it you will need to use a mapping such as [`_TASK_ALIASES`](https://github.com/huggingface/optimum-intel/blob/v1.7.3/optimum/intel/openvino/__init__.py#L46) (which can be moved to `optimum.intel.utils` as it can be useful for all subpackages).